homeassistant/components/yale_smart_alarm/config_flow.py

```diff
@@ -171,7 +171,7 @@ async def async_step_init(
             data_schema=vol.Schema(
                 {
                     vol.Optional(
-                        CONF_CODE, default=self.entry.options.get(CONF_CODE)
+                        CONF_CODE, default=self.entry.options.get(CONF_CODE, "")
```

Comment:
Why not default to `None` instead?
```suggestion
                        CONF_CODE, default=self.entry.options.get(CONF_CODE, None)
```

Reply:
Should return a string

Reply:
? That should not be needed, the above suggestion by DurgNosmis seems correct.

Reply:
which is the original code... 🤔

Reply:
```suggestion
                        CONF_CODE,
                        default={
                            "suggested_value": self.entry.options.get(
                               CONF_CODE
                            )
                        },
```

Reply:
As noticed in the issue if you simply skip this field then it tries to return None which is not accepted as it expects a string in return when saving from the form. I would assume given it's Optional it would simply take it but apparently not so quick fix just return a string.

Reply:
See my suggestion above, it adds the suggested value instead.

Reply:
Isn't that for description and not for default?
I tried and it just puts an object in the field instead.
Also first time you run the optionsflow it doesn't have a value hence we need to store something so I can of course change so it also allows to store "None" in the dict but is that better than just to provide an empty string?

Reply:
> I tried and it just puts an object in the field instead.

Are you sure you did it right? Please search the code base for "suggested_value" in config_flow.py files and you'll see this is a common pattern.

Reply:
It's description and not default but much better option to use so new commit coming this way now.